File: 0738-monotone-increasing-digits/0738-monotone-increasing-digits.java
class Solution {
    public int monotoneIncreasingDigits(int n) {
        String s = Integer.toString(n);
        char[] chars = s.toCharArray();
        int change = 0;
        for(int i = chars.length-1; i >= 1; --i) {
            if(chars[i] < chars[i-1]) {
                chars[i-1]--;
                change = i;
            }
        }
        
        if(change == 0) return n;
        for(int i = change; i < chars.length; ++i)
            chars[i] = '9';
        return Integer.parseInt(new String(chars));
    }
}

File: 0738-monotone-increasing-digits/0738-monotone-increasing-digits.py
class Solution(object):
    def monotoneIncreasingDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        s = list(str(n))
        change = 0
        for i in range(len(s)-1, 0, -1):
            if s[i] < s[i-1]:
                change = i
                s[i-1] = str(int(s[i-1])-1)        
        if change == 0: return n
        for i in range(change, len(s)):
            s[i] = '9'
        return int(''.join(s))
       
